src/qvimagecore.cpp

```diff
@@ -258,18 +261,17 @@ void QVImageCore::updateFolderInfo()
     if (!currentFileDetails.fileInfo.isFile())
         return;
 
+    currentFileDetails.folderFileInfoList = getCompatibleFiles();
+
     QPair<QString, uint> dirInfo = {currentFileDetails.fileInfo.absoluteDir().path(),
-                                    currentFileDetails.fileInfo.dir().count()};
+                                    currentFileDetails.folderFileInfoList.count()};
     // If the current folder changed since the last image, assign a new seed for random sorting
```

Comment:
Calling `fileInfo.dir().count()` results in `QDir` populating the entries behind the scenes. Instead, I'm just calling `getCompatibleFiles()` first and using its count to avoid the redundant file system access.